python/lsst/meas/extensions/scarlet/source.py

```diff
@@ -15,26 +15,31 @@ class LsstSource(ExtendedSource):
     default initialization and update constraints for general sources in
     LSST images.
     """
-    def __init__(self, peak, scene, observations, bg_rms, bbox, obs_idx=0,
-                 thresh=1, symmetric=False, monotonic=True, center_step=5,
-                 **component_kwargs):
+    def __init__(self, frame, peak, observation, bgRms, bbox,
+                 thresh=1, symmetric=True, monotonic=True, centerStep=5,
+                 pointSource=False, **componentKwargs):
         xmin = bbox.getMinX()
         ymin = bbox.getMinY()
-        sky_coord = np.array([peak.getIy()-ymin, peak.getIx()-xmin], dtype=int)
-        try:
-            super().__init__(sky_coord, scene, observations, bg_rms, obs_idx, thresh,
-                             symmetric, monotonic, center_step, **component_kwargs)
-        except SourceInitError:
-            # If the source is too faint for background detection, initialize
-            # it as a PointSource
-            PointSource.__init__(self, sky_coord, scene, observations, symmetric, monotonic,
-                                 center_step, **component_kwargs)
+        center = np.array([peak.getIy()-ymin, peak.getIx()-xmin], dtype=int)
+        initialized = False
+        if not pointSource:
+            try:
+                super().__init__(frame, center, observation, bgRms, thresh,
+                                 symmetric, monotonic, centerStep, **componentKwargs)
+                initialized = True
+            except SourceInitError:
+                # If the source is too faint for background detection,
+                # initialize it as a PointSource
+                pass
+        if not initialized:
```

Comment:
Should this happen if the pointSource is set to False?

Reply:
I see the above comment, but am just wondering the logic, should there not just be an error thrown or something?

Reply:
Part of this might be the nomenclature. Both `ExtendedSource` and `PointSource` are used to model extended sources, they just differ in their initialization. An `ExtendedSource` uses a multiband coadd to attempt to initialize large extended sources but requires a certain signal to noise level to function properly. Very faint sources barely above the background will not have enough flux to meet this requirement, so we initialize them as a point source but they are still allowed to grow.